java/arcs/android/storage/database/DatabaseCounters.kt

```diff
@@ -0,0 +1,118 @@
+package arcs.android.storage.database
+
+/** Collection of constants for use with [DatabasePerformanceStatistics] for [DatabaseImpl]. */
+object DatabaseCounters {
+    /* Entity-related Counters */
+    const val INSERTUPDATE_ENTITY = "insertUpdate_entity"
+    const val GET_ENTITY = "get_entity"
+    const val GET_ENTITY_STORAGEKEY_ID = "get_entity_storageKey_id"
+    const val GET_ENTITY_FIELDS = "get_entity_fields"
+    const val GET_ENTITY_REFERENCE = "get_entity_reference"
+    const val GET_ENTITY_TYPE_BY_STORAGEKEY = "get_entity_type_by_storageKey"
+    const val GET_ENTITY_FIELD_VALUES = "get_entity_field_values"
+    const val GET_ENTITY_FIELD_VALUE_PRIMITIVE = "get_entity_field_value_primitive"
+    const val ENTITY_SCHEMA_CACHE_HIT = "entity_schema_cache_hit"
+    const val ENTITY_SCHEMA_CACHE_MISS = "entity_schema_cache_miss"
+    const val INSERT_ENTITY_TYPE_ID = "insert_entity_type_id"
+    const val INSERT_ENTITY_FIELD = "insert_entity_field"
+    const val INSERT_ENTITY_REFERENCE = "insert_entity_reference"
+    const val INSERT_ENTITY_STORAGEKEY = "insert_entity_storageKey"
+    const val INSERT_ENTITY_RECORD = "insert_entity_record"
+    const val UPDATE_ENTITY_FIELD_VALUE = "update_entity_field_value"
+
+    /* Collection-related Counters */
+    const val INSERTUPDATE_COLLECTION = "insertUpdate_collection"
+    const val GET_COLLECTION = "get_collection"
+    const val GET_COLLECTION_ID = "get_collection_id"
+    const val GET_COLLECTION_ENTRIES = "get_collection_entries"
+    const val INSERT_COLLECTION_RECORD = "insert_collection_record"
+    const val INSERT_COLLECTION_STORAGEKEY = "insert_collection_storageKey"
+    const val DELETE_COLLECTION_ENTRIES = "delete_collection_entries"
+    const val INSERT_COLLECTION_ENTRY = "insert_collection_entry"
+
+    /* Singleton-related Counters */
+    const val INSERTUPDATE_SINGLETON = "insertUpdate_singleton"
+    const val GET_SINGLETON = "get_singleton"
+    const val GET_SINGLETON_ID = "get_singleton_id"
+    const val GET_SINGLETON_ENTRIES = "get_singleton_entries"
+    const val INSERT_SINGLETON_RECORD = "insert_singleton_record"
+    const val INSERT_SINGLETON_STORAGEKEY = "insert_singleton_storageKey"
+    const val DELETE_SINGLETON_ENTRY = "delete_singleton_entries"
+    const val INSERT_SINGLETON_ENTRY = "insert_singleton_entry"
+
+    /* Primitive Counters */
+    const val GET_BOOLEAN_VALUE_ID = "get_boolean_value_id"
+    const val GET_TEXT_VALUE_ID = "get_text_value_id"
+    const val GET_NUMBER_VALUE_ID = "get_number_value_id"
+    const val INSERT_TEXT_VALUE = "insert_text_value"
+    const val INSERT_NUMBER_VALUE = "insert_number_value"
+    const val GET_PRIMITIVE_VALUE_BOOLEAN = "get_primitive_value_boolean"
+    const val GET_PRIMITIVE_VALUE_TEXT = "get_primitive_value_text"
+    const val GET_PRIMITIVE_VALUE_NUMBER = "get_primitive_value_number"
+
+    /** [Array] of counter names for [DatabaseImpl.insertOrUpdate]. */
+    val INSERT_UPDATE_COUNTERS = arrayOf(
```

Comment:
iiuc, all counters at the same category (INSERT_UPDATE_COUNTERS, GET_COUNTERS, or DELETE_COUNTERS) will be updated altogether per time()/timeSuspending().
If so, do we want a finer granularity of categorization?
i.e. counting on collection-related operations affects counting on singleton-related ops and vice versa.

Reply:
That's a good idea. I'll make a follow-up bug to see how the counters are doing after a while, and re-visit.